### PR TITLE
Remove references to deprecated endpoints

### PR DESCRIPTION
**Pull request by FlorentinD**

### CODE REVIEW

File: doc/modules/ROOT/pages/common-datasets.adoc [hunk 1]
```diff
@@ -171,7 +171,7 @@ If the dataset is already found in the `dataset_root_path` however, an internet
 All OGB datasets are loaded into GDS with orientation `DIRECTED`.
 However, some of the datasets are _effectively undirected_ in the sense that for each relationship, one in the opposite direction between the same nodes also exists.
 The https://ogb.stanford.edu/[Open Graph Benchmark] website informs us on for which of the datasets this is the case.
-If you want the a projected graph to be `UNDIRECTED` in the Neo4j GDS sense of the word, you can modify it with https://neo4j.com/docs/graph-data-science/current/graph-catalog-relationship-ops/#catalog-graph-relationship-to-undirected-example[`gds.beta.graph.relationships.toUndirected`].
+If you want the a projected graph to be `UNDIRECTED` in the Neo4j GDS sense of the word, you can modify it with https://neo4j.com/docs/graph-data-science/current/graph-catalog-relationship-ops/#catalog-graph-relationship-to-undirected-example[`gds.graph.relationships.toUndirected`].
```
Comment (Mats-SX) [inline doc/modules/ROOT/pages/common-datasets.adoc:174]:
```suggestion
If you want a projected graph to be `UNDIRECTED` in the Neo4j GDS sense of the word, you can modify it with https://neo4j.com/docs/graph-data-science/current/graph-catalog-relationship-ops/#catalog-graph-relationship-to-undirected-example[`gds.graph.relationships.toUndirected`].
```

File: doc/modules/ROOT/pages/graph-object.adoc [hunk 2]
```diff
@@ -70,8 +70,8 @@ For users who are https://neo4j.com/docs/graph-data-science/current/management-o
 In addition to those aforementioned there are five more methods that create graph objects:
 
 * `gds.graph.project.cypher` (This is the Legacy Cypher projection, see <<cypher-projection>> for the new Cypher projection)
-* `gds.beta.graph.subgraph`
-* `gds.beta.graph.generate`
+* `gds.graph.subgraph`
```
Comment (Mats-SX) [inline doc/modules/ROOT/pages/graph-object.adoc:73]:
```suggestion
* `gds.graph.filter`
```

File: doc/modules/ROOT/pages/tutorials/community-detection.adoc [hunk 3]
```diff
@@ -202,7 +202,7 @@ For our further analysis we will work only with that subgraph.
 
 [source, python, role=no-test]
 ----
-largest_component_graph, _ = gds.beta.graph.project.subgraph(
+largest_component_graph, _ = gds.graph.project.subgraph(
```
Comment (Mats-SX) [inline doc/modules/ROOT/pages/tutorials/community-detection.adoc:205]:
```suggestion
largest_component_graph, _ = gds.graph.filter(
```

File: doc/modules/ROOT/pages/tutorials/community-detection.adoc [hunk 4]
```diff
@@ -58,7 +58,7 @@ gds = GraphDataScience(NEO4J_URI, auth=NEO4J_AUTH)
 ----
 from graphdatascience.server_version.server_version import ServerVersion
 
-assert gds.server_version() >= ServerVersion(1, 8, 0)
+assert gds.server_version() >= ServerVersion(2, 6, 0)
```
Comment (Mats-SX) [inline doc/modules/ROOT/pages/tutorials/community-detection.adoc:61]:
```suggestion
assert gds.server_version() >= ServerVersion(2, 5, 0)
```
https://github.com/neo-technology/graph-analytics/blob/293a1ba53b88aa9e138f085ac32fbcd574668f7e/release/archive/2.5.0.md#L149

Comment (Mats-SX) [reply doc/modules/ROOT/pages/tutorials/community-detection.adoc:61]: ah these are the generated tutorials. will comment on the notebook.

File: examples/community-detection.ipynb [hunk 5]
```diff
@@ -94,7 +94,7 @@
    "source": [
     "from graphdatascience.server_version.server_version import ServerVersion\n",
     "\n",
-    "assert gds.server_version() >= ServerVersion(1, 8, 0)"
+    "assert gds.server_version() >= ServerVersion(2, 6, 0)"
```
Comment (Mats-SX) [inline examples/community-detection.ipynb:97]:
```suggestion
    "assert gds.server_version() >= ServerVersion(2, 5, 0)"
```
https://github.com/neo-technology/graph-analytics/blob/293a1ba53b88aa9e138f085ac32fbcd574668f7e/release/archive/2.5.0.md#L149

File: examples/community-detection.ipynb [hunk 6]
```diff
@@ -322,7 +322,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "largest_component_graph, _ = gds.beta.graph.project.subgraph(\n",
+    "largest_component_graph, _ = gds.graph.project.subgraph(\n",
```
Comment (Mats-SX) [inline examples/community-detection.ipynb:325]:
```suggestion
    "largest_component_graph, _ = gds.graph.filter(\n",
```

File: examples/heterogeneous-node-classification-with-hashgnn.ipynb [hunk 7]
```diff
@@ -282,7 +282,7 @@
    "source": [
     "# Add a HashGNN node property step to the pipeline\n",
     "_ = pipe.addNodeProperty(\n",
-    "    \"beta.hashgnn\",\n",
+    "    \"hashgnn\",\n",
```
Comment (Mats-SX) [inline examples/heterogeneous-node-classification-with-hashgnn.ipynb:285]: This notebook has a text at the top which says GDS must be `1.6`. With this change it must be `2.5`.

Comment (Mats-SX) [reply examples/heterogeneous-node-classification-with-hashgnn.ipynb:285]: https://github.com/neo-technology/graph-analytics/blob/293a1ba53b88aa9e138f085ac32fbcd574668f7e/release/archive/2.5.0.md#L207

File: examples/import-sample-export-gnn.ipynb [hunk 8]
```diff
@@ -141,7 +141,7 @@
    "source": [
     "from graphdatascience.server_version.server_version import ServerVersion\n",
     "\n",
-    "assert gds.server_version() >= ServerVersion(2, 2, 0)"
+    "assert gds.server_version() >= ServerVersion(2, 6, 0)"
```
Comment (Mats-SX) [inline examples/import-sample-export-gnn.ipynb:144]:
```suggestion
    "assert gds.server_version() >= ServerVersion(2, 5, 0)"
```
I think 2.5 is enough to cover `gds.graph.relationships.stream` (`2.4` is enough for `rwr`). Don't feel strongly about it though.

File: examples/ml-pipelines-node-classification.ipynb [hunk 9]
```diff
@@ -620,7 +620,7 @@
     "id": "PNbRL2y84F6Q"
    },
    "source": [
-    "Instead of streaming the results, the prediction can be run in `mutate` mode to be more performant, especially when the predicted values are used multiple times. The predicted nodes can be retrieved using the `streamNodeProperty` method with the `UnclassifiedPaper` class."
+    "Instead of streaming the results, the prediction can be run in `mutate` mode to be more performant, especially when the predicted values are used multiple times. The predicted nodes can be retrieved using the `nodeProperty.stream` method with the `UnclassifiedPaper` class."
```
Comment (Mats-SX) [inline examples/ml-pipelines-node-classification.ipynb:623]: nice

File: examples/ml-pipelines-node-classification.ipynb [hunk 10]
```diff
@@ -113,7 +113,7 @@
    "source": [
     "from graphdatascience.server_version.server_version import ServerVersion\n",
     "\n",
-    "assert gds.server_version() >= ServerVersion(2, 2, 0)"
+    "assert gds.server_version() >= ServerVersion(2, 6, 0)"
```
Comment (Mats-SX) [inline examples/ml-pipelines-node-classification.ipynb:116]:
```suggestion
    "assert gds.server_version() >= ServerVersion(2, 5, 0)"
```

File: examples/node-regression-with-subgraph-and-graph-sample.ipynb [hunk 11]
```diff
@@ -428,7 +428,7 @@
     "\n",
     "# Third, we project a subgraph for each component\n",
     "component_graphs = [\n",
-    "    gds.beta.graph.project.subgraph(\n",
+    "    gds.graph.project.subgraph(\n",
```
Comment (Mats-SX) [inline examples/node-regression-with-subgraph-and-graph-sample.ipynb:431]:
```suggestion
    "    gds.graph.filter(\n",
```

File: examples/node-regression-with-subgraph-and-graph-sample.ipynb [hunk 12]
```diff
@@ -121,7 +121,7 @@
    "source": [
     "from graphdatascience.server_version.server_version import ServerVersion\n",
     "\n",
-    "assert gds.server_version() >= ServerVersion(2, 1, 0)"
+    "assert gds.server_version() >= ServerVersion(2, 6, 0)"
```
Comment (Mats-SX) [inline examples/node-regression-with-subgraph-and-graph-sample.ipynb:124]:
```suggestion
    "assert gds.server_version() >= ServerVersion(2, 5, 0)"
```

**Review by Mats-SX — CHANGES_REQUESTED** [review]
Very nice to get this updated. Some version assertions a bit stronger than they need to, and one more needs to be updated. Also, `graph.filter` is the final name for `subgraph` and must be used.

**Comment by FlorentinD** [comment]
@Mats-SX thanks for the good feedback.
I could find another reference to a non-existing `gds.grpah.project.subgraph` in our reference docs as well 👀

File: doc/modules/ROOT/pages/tutorials/centrality-algorithms.adoc [hunk 13]
```diff
@@ -56,7 +56,7 @@ gds = GraphDataScience(NEO4J_URI, auth=NEO4J_AUTH)
 
 [source, python, role=no-test]
 ----
-from graphdatascience.server_version.server_version import ServerVersion
```
Comment (Mats-SX) [inline doc/modules/ROOT/pages/tutorials/centrality-algorithms.adoc:59]: beautiful

**Pull request merged**